File: src/main/java/social/alone/server/event/controller/MyEventListApi.java
package social.alone.server.event.controller;


import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import social.alone.server.auth.oauth2.user.CurrentUser;
import social.alone.server.event.Event;
import social.alone.server.event.service.EventSearchService;
import social.alone.server.event.type.EventQueryParams;
import social.alone.server.user.User;

import javax.validation.Valid;
import java.util.Optional;

@Controller
@RequestMapping(value = "/api/events/my")
@RequiredArgsConstructor
public class MyEventListApi {

    private final EventSearchService eventSearchService;

    @GetMapping("/upcoming")
    public ResponseEntity upcoming(
            final Pageable pageable,
            @CurrentUser final User user,
            @Valid final EventQueryParams eventQueryParams
    ) {
        if (user == null){
            return ResponseEntity.badRequest().build();
        }

        Page<Event> page = this.eventSearchService.findAllBy(
                pageable,
                Optional.ofNullable(user),
                eventQueryParams
        );
        return ResponseEntity.ok(page);
    }

    @GetMapping("/past")
    public ResponseEntity past(
            final Pageable pageable,
            @CurrentUser final User user,
            @Valid final EventQueryParams eventQueryParams
    ) {

        if (user == null){
            return ResponseEntity.badRequest().build();
        }

        Page<Event> page = this.eventSearchService.findAllBy(
                pageable,
                Optional.ofNullable(user),
                eventQueryParams
        );
        return ResponseEntity.ok(page);
    }

}

File: src/main/kotlin/social/alone/server/event/controller/MyEventListApi.kt
package social.alone.server.event.controller


import org.springframework.data.domain.Pageable
import org.springframework.http.ResponseEntity
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import social.alone.server.auth.oauth2.user.CurrentUser
import social.alone.server.event.service.EventSearchService
import social.alone.server.event.type.EventQueryParams
import social.alone.server.user.domain.User
import javax.validation.Valid

@Controller
@RequestMapping(value = ["/api/events/my"])
class MyEventListApi(private val eventSearchService: EventSearchService) {

    @GetMapping("/upcoming")
    fun upcoming(
            pageable: Pageable,
            @CurrentUser user: User?,
            @Valid eventQueryParams: EventQueryParams
    ): ResponseEntity<*> {
        if (user == null) {
            return ResponseEntity.badRequest().build<Any>()
        }

        val page = this.eventSearchService!!.findAllBy(
                pageable,
                eventQueryParams
        )
        return ResponseEntity.ok(page)
    }

    @GetMapping("/past")
    fun past(
            pageable: Pageable,
            @CurrentUser user: User?,
            @Valid eventQueryParams: EventQueryParams
    ): ResponseEntity<*> {

        if (user == null) {
            return ResponseEntity.badRequest().build<Any>()
        }

        val page = this.eventSearchService!!.findAllBy(
                pageable,
                eventQueryParams
        )
        return ResponseEntity.ok(page)
    }

}
